packages/ui-modal/src/Modal/index.tsx

```diff
@@ -211,7 +240,7 @@ class Modal extends Component<ModalProps, ModalState> {
         open
         label={label}
         defaultFocusElement={this.defaultFocusElement}
-        shouldCloseOnDocumentClick={shouldCloseOnDocumentClick}
```

Comment:
`shouldCloseOnDocumentClick` is set to `false` to disable the logic of the `FocusRegion`